### PR TITLE
BIP 148: Mandatory activation of segwit deployment

### PR DESCRIPTION
**Pull request by shaolinfry**

### CODE REVIEW

**Comment by ntom** [comment]
Compulsory activation ? wtf ?   
We're not deciding or voting anything now, no signalling..  someone, somewhere is just forcing this by law ? r u kidding me ? get real. fucking ponker

**Comment by is55555** [comment]
An effort to provide a PoW contingency plan is also being started, to provide back up for any eventuality in this transition.

https://bitcointalk.org/index.php?topic=1833391

**Comment by seweso** [comment]
What might be smart to add is some kind of depth at which blocks are accepted. Maybe also make it configurable. I really like the idea of consensus emerging from the network itself. It's like decentralised consensus finding.

**Comment by dabura667** [comment]
> What might be smart to add is some kind of depth at which blocks are accepted. Maybe also make it configurable. I really like the idea of consensus emerging from the network itself. It's like decentralised consensus finding.

I see what you did there.

**Comment by farukuzun** [comment]
Why did you choose between October 1st 2017 and November 15th 2017? 
I don't think this bip acceptable but core developers will come up with something better to prevent miners to takeover the network I hope. So some memecoin backed by a couple rich can't and won't try to replace it's name with bitcoin in the future. Since bitcoin is the protocol, developers can force laws like this. They are good for decentralization.

**Comment by aayanl** [comment]
This is just a hardfork, while you're at it, put in a block size increase there too

**Comment by farukuzun** [comment]
@aayanl segwit already has it

**Comment by pdaian** [comment]
I left a clarifying question on the original proposal here: https://gist.github.com/shaolinfry/743157b0b1ee14e1ddc95031f1057e4c#gistcomment-2028169

No answer between when I left it four days ago and when this was submitted as a PR today.

The rationale for the choice of a 2-DoS score is still unclear in the proposal.

File: bip-segwit-flagday.mediawiki [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+*[[bip-0009.mediawiki|BIP9 Version bits with timeout and delay]]
+*[[bip-0016.mediawiki|BIP16 Pay to Script Hash]]
+*[[bip-0141.mediawiki|BIP141 Segregated Witness (Consensus layer)]]
+*[[bip-0143.mediawiki|BIP143 Transaction Signature Verification for Version 0 Witness Program]]
+*[[bip-0147.mediawiki|BIP147 Dealing with dummy stack element malleability]]
+*[https://bitcoincore.org/en/2016/01/26/segwit-benefits/ Segwit benefits]
+
+==Copyright==
+
+This document is placed in the public domain.
```
Comment (luke-jr) [inline bip-segwit-flagday.mediawiki]: This doesn't match the header (and PD is not an acceptable "license" anymore).

**Comment by luke-jr** [comment]
As a reminder to commenters here: discussion of the actual proposal should occur on the bitcoin-dev mailing list, not on the pull request, which will be merged/closed as soon as the process issues are resolved, and without any consideration for the merits of or flaws in the idea.

**Comment by Cocosoft** [comment]
> This is just a hardfork, while you're at it, put in a block size increase there too

No, this is not a hardfork, old legacy nodes will still follow the UASF/Softfork chain, should it become the one with most hashing power.

In a hardfork, old nodes would never accept the hardfork chain.

---

Perhaps requiring ~51% miners support would make the proposal more safer?

**Comment by shaolinfry** [comment]
@farukuzun Please see the [mailing list discussion](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2017-March/013714.html)

@ntom P2SH was activated by flag day also, see [0.6.0 release](https://github.com/bitcoin/bitcoin/blob/v0.6.0/src/main.cpp#L1281-L1283)

@luke-jr Done. Unsure why Travis is failing.

**Comment by shaolinfry** [comment]
@luke-jr OK Travis errors fixed.

**Comment by WriteCodeEveryday** [comment]
This is exactly what we need. It shows we at Blockstream own Bitcoin and if you don't align with our vision, you will be crushed like cockroaches. 

I wonder how many percentage points off of the original 95% we'll be at when this activates and hard forks those nodes and miners off the network.

File: bip-0148.mediawiki [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+*[[bip-0009.mediawiki|BIP9 Version bits with timeout and delay]]
+*[[bip-0016.mediawiki|BIP16 Pay to Script Hash]]
+*[[bip-0141.mediawiki|BIP141 Segregated Witness (Consensus layer)]]
+*[[bip-0143.mediawiki|BIP143 Transaction Signature Verification for Version 0 Witness Program]]
+*[[bip-0147.mediawiki|BIP147 Dealing with dummy stack element malleability]]
+*[https://bitcoincore.org/en/2016/01/26/segwit-benefits/ Segwit benefits]
+
+==Copyright==
+
+This document dual licensed as BSD-3-Clause and CC0-1.0.
```
Comment (dhimmel) [inline bip-0148.mediawiki]: `BSD 3-Clause` and `CC0 1.0`: without dashes. 

Add hyperlinks to authoritative license deeds.

This document **is**

**Comment by AllanDoensen** [comment]
No miner is going to run code that will put him on a minority chain. I hope blockstream does this real soon cause it will bury blockstream/core forever.

**Comment by afk11** [comment]
Concept ACK. Better to trigger activation for the >80% of nodes prepared to enforce the rule already.

**Pull request merged**